src/tiles/tiles.ts

```diff
@@ -1,244 +1,80 @@
-  );
-
-  if (clamp) {
-    // clamp the values in cases where a metatile may extend tile bounding box beyond the bounding box
-    // of the tile matrix
-    const { min: tileMatrixBoundingBoxMin, max: tileMatrixBoundingBoxMax } = tileMatrixToBoundingBox(tileMatrix);
-    return new BoundingBox([
-      clampValues(boundingBox.min.lon, tileMatrixBoundingBoxMin.lon, tileMatrixBoundingBoxMax.lon),
-      clampValues(boundingBox.min.lat, tileMatrixBoundingBoxMin.lat, tileMatrixBoundingBoxMax.lat),
-      clampValues(boundingBox.max.lon, tileMatrixBoundingBoxMin.lon, tileMatrixBoundingBoxMax.lon),
-      clampValues(boundingBox.max.lat, tileMatrixBoundingBoxMin.lat, tileMatrixBoundingBoxMax.lat),
-    ]);
-  }
-
-  return boundingBox;
-}
-
-/**
- * Calculates a tile for a longitude, latitude and tile matrix
- * @param geoPoint point with longitude and latitude
- * @param tileMatrix tile matrix which the calculated tile belongs to
- * @param reverseIntersectionPolicy boolean value whether to reverse the intersection policy (in cases that the location is on the edge of the tile)
- * @param metatile size of a metatile
- */
-function geoCoordsToTile<T extends TileMatrixSet>(
-  geoPoint: GeoPoint,
-  tileMatrix: ArrayElement<T['tileMatrices']>,
-  reverseIntersectionPolicy: boolean,
-  metatile = 1
-): Tile<T> {
-  const width = tileEffectiveWidth(tileMatrix) * metatile;
-  const height = tileEffectiveHeight(tileMatrix) * metatile;
-
-  const { min: tileMatrixBoundingBoxMin, max: tileMatrixBoundingBoxMax } = tileMatrixToBoundingBox(tileMatrix);
-
-  const {
-    identifier: { code: tileMatrixId },
-    cornerOfOrigin = 'topLeft',
-  } = tileMatrix;
-
-  const x = (geoPoint.lon - tileMatrixBoundingBoxMin.lon) / width;
-  const y = (cornerOfOrigin === 'topLeft' ? tileMatrixBoundingBoxMax.lat - geoPoint.lat : geoPoint.lat - tileMatrixBoundingBoxMin.lat) / height;
+    const movingTileRanges: TileRange<T>[] = [];
+    intersectingPolygons.features
+      .map((polygon) => {
+        const boundingBox = new BoundingBox(bbox(polygon.geometry));
+        const { minTileCol, minTileRow, maxTileCol, maxTileRow } = boundingBox.toTileRange(tileMatrix, metatile);
+        return new TileRange(minTileCol, minTileRow, maxTileCol, maxTileRow, tileMatrixId, metatile);
+      })
+      .sort((a, b) => (width > height ? a.minTileRow - b.minTileRow : a.minTileCol - b.minTileCol))
+      .forEach((tileRange) => {
+        const lastTileRange = movingTileRanges.at(-1);
+
+        if (!lastTileRange) {
+          movingTileRanges.push(tileRange);
+          return;
+        }
+
+        const [prevMin, prevMax, currMin, currMax] =
+          width > height
+            ? [lastTileRange.minTileCol, lastTileRange.maxTileCol, tileRange.minTileCol, tileRange.maxTileCol]
+            : [lastTileRange.minTileRow, lastTileRange.maxTileRow, tileRange.minTileRow, tileRange.maxTileRow];
+
+        if (currMin - 1 <= prevMax) {
+          // merge with last
```

Comment:
yes it is possible if they have matcing size on at least one of athe axis

Reply:
outdated. currently `toTileMatrixLimits()` uses such a private function internally `mergeTileMatrixLimits()` perhaps with few changes it could become a public one